sdk/search/Azure.Search.Documents/src/Models/DataType.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+
+namespace Azure.Search.Documents.Models
+{
+    [CodeGenSchema("DataType")]
+    public readonly partial struct DataType
+    {
+#pragma warning disable CA1720 // Identifier contains type name
+        /// <summary>A <see cref="string"/> type.</summary>
+        [CodeGenSchemaMember("EdmString")]
+        public static DataType String { get; } = new DataType(StringValue);
+
+        /// <summary>An <see cref="int"/> type, or something that can convert to an <see cref="int"/>.</summary>
+        [CodeGenSchemaMember("EdmInt32")]
+        public static DataType Int32 { get; } = new DataType(Int32Value);
+
+        /// <summary>An <see cref="long"/> type, or something that can convert to an <see cref="long"/>.</summary>
+        [CodeGenSchemaMember("EdmInt64")]
+        public static DataType Int64 { get; } = new DataType(Int64Value);
+
+        /// <summary>A <see cref="double"/> type, or something that can convert to a <see cref="double"/>.</summary>
+        [CodeGenSchemaMember("EdmDouble")]
+        public static DataType Double { get; } = new DataType(DoubleValue);
+#pragma warning restore CA1720 // Identifier contains type name
+
+        /// <summary>A <see cref="bool"/> type.</summary>
+        [CodeGenSchemaMember("EdmBoolean")]
+        public static DataType Boolean { get; } = new DataType(BooleanValue);
+
+        /// <summary>A <see cref="System.DateTimeOffset"/> type, or <see cref="System.DateTime"/> converted to a <see cref="System.DateTimeOffset"/>.</summary>
+        [CodeGenSchemaMember("EdmDateTimeOffset")]
+        public static DataType DateTimeOffset { get; } = new DataType(DateTimeOffsetValue);
+
+        /// <summary>A geographic point type.</summary>
+        [CodeGenSchemaMember("EdmGeographyPoint")]
+        public static DataType GeographyPoint { get; } = new DataType(GeographyPointValue);
+
+        /// <summary>A complex type with child fields.</summary>
+        [CodeGenSchemaMember("EdmComplexType")]
+        public static DataType Complex { get; } = new DataType(ComplexValue);
+
+        /// <summary>
+        /// Gets a <see cref="DataType"/> representing a collection of <paramref name="type"/>.
+        /// </summary>
+        /// <param name="type">The type of collection.</param>
+        /// <returns>A <see cref="DataType"/> representing a collection of <paramref name="type"/>.</returns>
+        public static DataType Collection(DataType type) => $"Collection({type})";
```

Comment:
I've been thinking about this before you asked. There could be more validation (if we wanted to) if we detected collections, but we agreed at the archboard level for .NET "extensible enums" not to attach other data. While I was working on the standard on which we eventually agreed, I tried this in KV and you have to implement a lot more.

However, we could have a calculated property that checks for "Collection(" as a prefix, for example, and return true. As long as we don't call it in a tight loop or otherwise too often, that's probably fine. And it would be helpful in this case, because @brjohnstmsft confirmed that collections of collections are not possible.

Reply:
We could also consider just adding a `bool IsCollection` to DataType and making it richer than an extensible enum.  We can track this and address later.

Reply:
That's what I did for KV originally. It's still an extensible enum, just with more values that you have to parse out; thus, the guidance was adopted that to attach data you actually attach behaviors. As long as we don't have to determine if it's a collection, the `string.StartsWith` check wouldn't be expensive, which I believe would be the normal use case - especially if we expect people are normally using a `FieldBuilder` (TBA) for this and never interactiving with it normally.